File: exercism/javascript/hamming/hamming.js
class Hamming {
    compute(strand1, strand2){
        if(strand1.length !== strand2.length)
            throw Error('DNA strands must be of equal length.')
        
        return strand1
            .split('')
            .filter((x, i) => x !== strand2.split('')[i])
            .length
    }
}
module.exports = Hamming

//// Solution by nelsonkhan 

File: exercism/ruby/hamming/hamming.rb
class Hamming
  def self.compute(dnaA, dnaB)
    dnaA.length != dnaB.length ? compute : nil
    hamming = 0
    i = 0
    dnaA.each_char do |n|
      if n != dnaB[i]
        hamming += 1
      end
      i += 1
    end
    return hamming
  end
end
